In [1]:
%pip install wikipedia

  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Created wheel for wikipedia: filename=wikipedia-1.4.0-py3-none-any.whl size=11756 sha256=8476254fb998055dc417355358c1e23e3d11b4361c4aa8183efbccedeeac8532
  Stored in directory: /home/codespace/.cache/pip/wheels/63/47/7c/a9688349aa74d228ce0a9023229c6c0ac52ca2a40fe87679b8
Successfully built wikipedia

[notice] A new release of pip is available: 25.3 -> 26.0.1
[notice] To update, run: python3 -m pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [2]:
import wikipedia

In [3]:
wiki_page=wikipedia.page(r'ISO_3166-1')

In [4]:
wiki_page

<WikipediaPage 'ISO 3166'>

In [5]:
wikipedia.page??

Signature:
wikipedia.page(
    title=None,
    pageid=None,
    auto_suggest=True,
    redirect=True,
    preload=False,
)
Source:   
def page(title=None, pageid=None, auto_suggest=True, redirect=True, preload=False):
  '''
  Get a WikipediaPage object for the page with title `title` or the pageid
  `pageid` (mutually exclusive).

  Keyword arguments:

  * title - the title of the page to load
  * pageid - the numeric pageid of the page to load
  * auto_suggest - let Wikipedia find a valid page title for the query
  * redirect - allow redirection without raising RedirectError
  * preload - load content, summary, images, references, and links during initialization
  '''

  if title is not None:
    if auto_suggest:
      results, suggestion = search(title, results=1, suggestion=True)
      try:
        title = suggestion or results[0]
      except IndexError:
        # if there is no suggestion or search results, the page doesn't exist
        raise PageError(title)
    return Wikipedia

In [6]:
wikipedia.search(r'ISO_3166-1', results=1, suggestion=True)

(['ISO 3166-1'], 'iso 3166')

In [7]:
wiki_page_init=wikipedia.WikipediaPage(r'ISO_3166-1')

In [8]:
wiki_page_init

<WikipediaPage 'ISO 3166-1'>

In [9]:
wiki_page2=wikipedia.page(r'ISO_3166-1',auto_suggest = False)

In [10]:
wiki_page2

<WikipediaPage 'ISO 3166-1'>

In [11]:
file_dump=open('wiki_file_dump.html','w')
file_dump.write(wiki_page2.html())
file_dump.close()

```html
<table class="wikitable sortable">
<tbody><tr>
<th>Short name <br /> upper case <br /> in ISO 3166
</th>
<th>Short name lower case <br /> in ISO 3166
<hr />Full name
</th>
<th>Listed as independent <br /> in ISO 3166
</th>
<th>Local <br /> short name
</th>
<th class="unsortable">Dispute
</th>
<th>Link to ISO 3166-2
</th></tr>
<tr>
<td>AFGHANISTAN</td>
<td>Afghanistan<hr />the <a href="/wiki/Islamic_Republic_of_Afghanistan" title="Islamic Republic of Afghanistan">Islamic Republic of Afghanistan</a></td>
<td style="background:#9EFF9E;color:black;vertical-align:middle;text-align:center;" class="table-yes">Yes<sup id="cite&#95;ref-6" class="reference"><a href="#cite_note-6"><span class="cite-bracket">&#91;</span>6<span class="cite-bracket">&#93;</span></a></sup>
</td>
<td><style data-mw-deduplicate="TemplateStyles:r1126788409">.mw-parser-output .plainlist ol,.mw-parser-output .plainlist ul{line-height:inherit;list-style:none;margin:0;padding:0}.mw-parser-output .plainlist ol li,.mw-parser-output .plainlist ul li{margin-bottom:0}</style><div class="plainlist">
<ul><li><span title="Pashto-language romanization"><i lang="ps-Latn">Afghānistān</i></span> (<a href="/wiki/Pashto" title="Pashto">ps</a>)</li>
<li><span title="Persian-language romanization"><i lang="fa-Latn">Afghānistān</i></span> (<a href="/wiki/Persian_language" title="Persian language">fa</a>)</li></ul>
</div>
</td>
<td>Under the <a href="/wiki/Recognition_of_the_Islamic_Emirate_of_Afghanistan" title="Recognition of the Islamic Emirate of Afghanistan">unrecognized government</a> of the Islamic Emirate of Afghanistan</td>
<td><a href="/wiki/ISO_3166-2:AF" title="ISO 3166-2:AF">ISO 3166-2:AF</a>
</td></tr>
```
renders as 
<table class="wikitable sortable">
<tbody><tr>
<th>Short name <br /> upper case <br /> in ISO 3166
</th>
<th>Short name lower case <br /> in ISO 3166
<hr />Full name
</th>
<th>Listed as independent <br /> in ISO 3166
</th>
<th>Local <br /> short name
</th>
<th class="unsortable">Dispute
</th>
<th>Link to ISO 3166-2
</th></tr>
<tr>
<td>AFGHANISTAN</td>
<td>Afghanistan<hr />the <a href="/wiki/Islamic_Republic_of_Afghanistan" title="Islamic Republic of Afghanistan">Islamic Republic of Afghanistan</a></td>
<td style="background:#9EFF9E;color:black;vertical-align:middle;text-align:center;" class="table-yes">Yes<sup id="cite&#95;ref-6" class="reference"><a href="#cite_note-6"><span class="cite-bracket">&#91;</span>6<span class="cite-bracket">&#93;</span></a></sup>
</td>
<td><style data-mw-deduplicate="TemplateStyles:r1126788409">.mw-parser-output .plainlist ol,.mw-parser-output .plainlist ul{line-height:inherit;list-style:none;margin:0;padding:0}.mw-parser-output .plainlist ol li,.mw-parser-output .plainlist ul li{margin-bottom:0}</style><div class="plainlist">
<ul><li><span title="Pashto-language romanization"><i lang="ps-Latn">Afghānistān</i></span> (<a href="/wiki/Pashto" title="Pashto">ps</a>)</li>
<li><span title="Persian-language romanization"><i lang="fa-Latn">Afghānistān</i></span> (<a href="/wiki/Persian_language" title="Persian language">fa</a>)</li></ul>
</div>
</td>
<td>Under the <a href="/wiki/Recognition_of_the_Islamic_Emirate_of_Afghanistan" title="Recognition of the Islamic Emirate of Afghanistan">unrecognized government</a> of the Islamic Emirate of Afghanistan</td>
<td><a href="/wiki/ISO_3166-2:AF" title="ISO 3166-2:AF">ISO 3166-2:AF</a>
</td></tr></tbody></table>

No sort arrows when raw html rendered outside wikipedia website. No funny non-printable characters.